pages/lazer/how-lazer-works.mdx

```diff
@@ -1,5 +1,73 @@
 
-We are working on writing a detailed technical overview how Lazer works. Details will be added here as they are completed.
+## System Services
+
+The architecture consists of five main types of services that work together to provide ultra-low-latency data to consumers.
+Each service has multiple instances running to ensure high availability and low latency.
+
+### INSERT DIAGRAM HERE
+
+### Publishers
+
+Publishers are the entities that provide market data to Lazer. They submit updates via authenticated WebSocket connections.
+Each publisher is configured with specific permissions defining which feeds they can update.
+
+### Relayers
+
+The Relayer service is the ingestion layer that receives and validates all incoming updates from publishers.
+
+**Key responsibilities:**
+
+- **Authentication**: Validates publisher access tokens and optional Ed25519 signatures.
+- **Validation**: Performs sanity checks on incoming updates by examining feed IDs, timestamps, and values to ensure data integrity and proper formatting.
+- **Rate limiting**: Enforces configurable limits on publisher updates.
+- **Message forwarding**: Publishes validated updates to an internal message queue.
+
+Douro Labs operates the relayer service for the Pyth Lazer network. It follows a strict, deterministic processing model:
+
+- **No price dropping outside of circuit breakers**: All validated updates are forwarded to the message queue without dropping any prices.
+- **FCFS processing**: Updates are processed on a first-come-first-served basis without prioritization.
+- **Deterministic operation**: The relayer operates according to its configured logic and does not deviate from it.
+
+This ensures reliable, predictable data flow from publishers to consumers.
+
+### Message Queue
+
+The system uses a distributed message queue for pub/sub messaging with stream persistence.
+This allows the system to be deployed in a multi-datacenter environment and ensures reliable message delivery between services.
+
+**Message ordering**: The message queue guarantees message ordering within a single stream, ensuring that updates from publishers are processed in the order they were received by the Relayer.
+This ordering guarantee is critical for maintaining consistent feed state across all aggregators.
+
+### Routers
+
+The Router is the real-time distribution layer that serves data to consumers.
+It embeds aggregation logic to compute median prices, confidence intervals (using interquartile range), and best bid/ask prices, funding rates, and more from multiple publisher inputs.
+
+**Key features:**
+
+- **WebSocket streaming**: Provides `/v1/stream` endpoint for real-time price updates
+- **HTTP REST API**: Offers `/v1/latest_price` for on-demand price queries
+- **Channel types**: Supports real-time and fixed-rate channels (1ms, 50ms, 200ms)
+- **Multi-chain support**: Generates on-chain payloads for Solana, EVM, and other chains
+
+#### Aggregation logic
```

Comment:
this is where deterministic behaviour is important, all the aggregators produce the exact same aggregate solely relying on the consistent stream of price updates.